libraries/Servo/src/Servo.cpp

```diff
@@ -84,21 +84,20 @@ void Servo::detach()
   if (_attached) {
     stopWaveform(_pin);
     _attached = false;
+    _valueUs = DEFAULT_NEUTRAL_PULSE_WIDTH;
     digitalWrite(_pin, LOW);
   }
 }
 
 void Servo::write(int value)
 {
-  // treat values less than 544 as angles in degrees (valid values in microseconds are handled as microseconds)
+  // treat values less than _minUs as angles in degrees (values equal or larger are handled as microseconds)
   if (value < _minUs) {
```

Comment:
what does the AVR Arduino do in this case?  Does it use the_minUs and _maxUs or does it use some other hardcoded values.
What I am getting at here, is the core issue your concerned about is that the angle set and get are calculated act differently (dangerously as you put it) than the standard servo library?

Reply:
@makuna have you looked at the ostensible default from Arduino and how nothing has been done there in 3.5 years over an active issue https://github.com/arduino-libraries/Servo/issues/3 ?
The "safety limits" in the ESP8266 impl are a local invention, btw, I don't see the AVR impl enforcing any such restriction, but the code is obfuscated, probably to conserve RAM (int8_t instead of int) at the expense of CPU time.
Now, what is that default behavior...

Reply:
@dok-net Let me be a little more clear.  Provide more details as to why deviating from the AVR/SAM/etc libraries is a good thing.  What research have you done?  Links to data to back up your statements.  
I wrote my own for AVR that uses a separate timer for each channel so NO ISRs happen; and as you state, I used pulses in the range of 1ms - 2ms with a 1.5ms center as defaults; but mine doesn't limit the user, it just gives them something to play with and lets them keep it in range.(https://github.com/Makuna/PwmServo/blob/master/PwmServo.h)

Reply:
At to what happens, here's the story again, this can happen to anyone with any servo, the results may be more or less severe, but the gist of it is, the defaults should be inside safe limits, finding out what the exact 0° and 180° values are for any given serve is a matter of calibration.

I inspected the sweep example, adopted the initialization code into my own sketch, started it, set a few values for the outputs, among them the servo - my device makes a little noise that can mask the servo's - and found the servo pressing against it's internal mechanical endstop, running hot while doing so, and before I knew it, it stopped working for good.
Mind you, this happens well below requesting 180° with the lib's defaults.

Reply:
I checked the Arduino issue on this and yes, it's been years without any response or motion, and in those years only had 2 thumbs up (one from dok-net, one from another user).  It seems unlikely to move, but frankly that's normal for Arduino stuff.

I'm not really able to find a meaningful datasheet for consumer-type servos.  Some websites report 500-2500 for certain models, while others are reporting 900-2200 for the same exact models.

That said, the only thing changing are the *defaults*, from a wider range to a (safer) narrower one.  `Attach`ing with specific mac/min values is still supported, and given the variability in the servos out there (aka clones) for real apps you're going to need an encoder or a manual tweaking anyway.  

@devyte, @d-a-v, any comments?